### PR TITLE
Humanise() tweaks rebased

### PR DESCRIPTION
**Pull request by richardbuckle**
Reduce precision a little bit when describing very large numbers (larger than 10 billion).
Tidy up numbers using scientific notation
Expand test coverage.

### CODE REVIEW

File: SpeechService/Translations.cs [hunk 1]
```diff
@@ -728,32 +729,57 @@ public static string Humanize(decimal? value)
...
+                    case 0:
+                        return Properties.Phrases.justover + " " + minus + number + order;
+                    case 1:
+                        return Properties.Phrases.over + " " + minus + number + order;
+                    case 2:
+                        return Properties.Phrases.wellover + " " + minus + number + order;
+                    case 3:
+                        return Properties.Phrases.onthewayto + " " + minus + number + andahalf + order;
+                    case 4:
+                        return Properties.Phrases.nearly + " " + minus + number + andahalf + order;
```
Comment (richardbuckle) [inline SpeechService/Translations.cs]: Some lacking of testing of the newly introduced cases, but more importantly I have to wonder if being this fine-grained on the next digit is justified.

File: SpeechService/Translations.cs [hunk 2]
```diff
@@ -678,6 +678,7 @@ public static string Humanize(decimal? value)
             int nextDigit;
             string order;
             int digits = (int)Math.Log10((double)value);
+            bool morePreciseDecimalValues = digits < 10;
```
Comment (richardbuckle) [inline SpeechService/Translations.cs]: Why this criterion, and why hard-coded? 
Also, why not just compare `value < 1e10` which would be much quicker (single machine op vs log10 call)?

File: SpeechService/Translations.cs [hunk 3]
```diff
@@ -688,37 +689,37 @@ public static string Humanize(decimal? value)
             else if (digits < 6)
             {
                 // Thousands
-                number = (int)(value / 1000);
+                number = (int)(value / 1E3M);
```
Comment (richardbuckle) [inline SpeechService/Translations.cs]: This is all good.

**Comment by Tkael** [comment]
> 
> 
> It's sound but what's the justification?

I'd prefer to do without the extra cognitive burden of knowing that a neutron star's density is 425 **and a half** trillion kg/m and I think the same principle basically would apply to any very large number.

File: SpeechService/Translations.cs (hunk 2 above)
Comment (Tkael) [reply SpeechService/Translations.cs]: I admit, this one's a bit of an arbitrary number. The thought is that once we're dealing with numbers larger than 10 billion, I don't really care to hear about half a billion, half a trillion, etc. and am looking for a way to reduce the cognitive burden.

**Comment by Hoodathunk** [comment]
Good engineering rule of thumb is 3 digits of accuracy, max. Anything more is overkill.

**Comment by Keeper1st** [comment]
That's partly why I initially suggested "point x" but I agree that if the initial amount is three digits, it isn't needed. In other words, "425 trillion", and "25 point 3 trillion" seems most natural.

**Comment by Tkael** [comment]
I think that "point x" is a little less human than we're going for here, though I appreciate the suggestion. :-)

**Comment by Tkael** [comment]
Closing this PR to reduce redundancy / confusion vs. #1414.

**Pull request closed without merging**